rust_icu_ubrk/src/lib.rs

```diff
@@ -0,0 +1,866 @@
+        Self::try_new_ustring(type_, &locale, &text)
+    }
+
+    /// Implements `ubrk_open`.
+    pub fn try_new_ustring(
+        type_: sys::UBreakIteratorType,
+        locale: &uloc::ULoc,
+        text: &ustring::UChar,
+    ) -> Result<Self, common::Error> {
+        let mut status = common::Error::OK_CODE;
+        // Clone text and get locale as a CString for break iterator to own.
+        let locale = locale.as_c_str();
+        let text = text.clone();
+        let rep = unsafe {
+            assert!(common::Error::is_ok(status));
+            versioned_function!(ubrk_open)(
+                type_,
+                locale.as_ptr(),
+                text.as_c_ptr(),
+                text.len() as i32,
+                &mut status,
+            )
+        };
+        common::Error::ok_or_warning(status)?;
+        assert_ne!(rep, 0 as *mut sys::UBreakIterator);
+        Ok(Self {
+            locale: Some(Rc::new(locale)),
+            rules: None,
+            binary_rules: None,
+            text: Rc::new(text),
+            rep: ptr::NonNull::new(rep).unwrap(),
+        })
+    }
+
+    /// Creates a new break iterator using the specified breaking rules.
+    ///
+    /// See the [ICU user guide](https://unicode-org.github.io/icu/userguide/boundaryanalysis/break-rules.html)
+    /// for rules syntax.
+    ///
+    /// Implements `ubrk_openRules`.
+    pub fn try_new_rules(
+        rules: &str,
+        text: &str,
+    ) -> Result<Self, common::Error> {
+        let rules = ustring::UChar::try_from(rules)?;
+        let text = ustring::UChar::try_from(text)?;
+        Self::try_new_rules_ustring(&rules, &text)
+    }
+
+    /// Implements `ubrk_openRules`.
+    pub fn try_new_rules_ustring(
+        rules: &ustring::UChar,
+        text: &ustring::UChar,
+    ) -> Result<Self, common::Error> {
+        let mut status = common::Error::OK_CODE;
+        let mut parse_status = common::NO_PARSE_ERROR;
+        // Clone text and rules for break iterator to own.
+        let rules = rules.clone();
+        let text = text.clone();
+        let rep = unsafe {
+            assert!(common::Error::is_ok(status));
+            versioned_function!(ubrk_openRules)(
+                rules.as_c_ptr(),
+                rules.len() as i32,
+                text.as_c_ptr(),
+                text.len() as i32,
+                &mut parse_status,
+                &mut status,
+            )
+        };
+        common::Error::ok_or_warning(status)?;
+        common::parse_ok(parse_status)?;
+        assert_ne!(rep, 0 as *mut sys::UBreakIterator);
+        Ok(Self {
+            locale: None,
+            rules: Some(Rc::new(rules)),
+            binary_rules: None,
+            text: Rc::new(text),
+            rep: ptr::NonNull::new(rep).unwrap(),
+        })
+    }
+
+    /// Creates a new break iterator using pre-compiled binary rules.
+    ///
+    /// Binary rules can be obtained with [`get_binary_rules`].
```

Comment:
I think that if you use the qualified name of `get_binary_rules`, it will automatically get linked and formatted with fixed spacing font.

Like so:

```
[UBreakIterator::get_binary_rules]
```

No need to change now, but it is something to be aware of and can make documenting a bit easier.